core/src/main/java/io/grpc/internal/GzipInflatingBuffer.java

```diff
@@ -287,6 +292,10 @@ int inflateBytes(byte[] b, int offset, int length) throws DataFormatException, Z
           throw new AssertionError("Invalid state: " + state);
       }
     }
+    // If we finished a gzip block, check if we have enough bytes to read another header
```

Comment:
Why? If we ignore that case won't we just notice the next time we try to read?

Reply:
Yes, this isn't necessary. But not having to do that extra read to detect stalled makes it simpler to handle end-of-stream in message deframer. Will check on the necessary changes in message deframer (if we dropped this case here) and update this comment.